augur/filter/__init__.py

```diff
@@ -68,6 +69,12 @@ def register_arguments(parser):
     output_group.add_argument('--output-metadata', help="metadata for strains that passed filters")
     output_group.add_argument('--output-strains', help="list of strains that passed filters (no header)")
     output_group.add_argument('--output-log', help="tab-delimited file with one row for each filtered strain and the reason it was filtered. Keyword arguments used for a given filter are reported in JSON format in a `kwargs` column.")
+    output_group.add_argument(
+        '--empty-results-reporting',
```

Comment:
This option doesn't fit under the "outputs" group:

> outputs:
>   possible representations of filtered data (at least one required)

Maybe it would fit better under the default "options" group?

```suggestion
    parser.add_argument(
        '--empty-results-reporting',
```

Reply:
> Maybe it would fit better under the default "options" group?

Hmm, I would look for options related to the outputs in the `outputs` group. 
I can reword the argument group description to make it clearer?

Reply:
> Though maybe the option name should use "output" instead of "results", e.g. --empty-output-reporting?

I specifically named the option with "results" since we have the `--output-log` option, which is not taken into account by the `--empty-results-reporting` option. That might just be me being nitpicky about words 🤷

Reply:
Updated description in force push:
```
outputs:
  options related to outputs, at least one of the possible representations
  of filtered data (--output, --output-metadata, --output-strains) is
  required

  --output OUTPUT, --output-sequences OUTPUT, -o OUTPUT
                        filtered sequences in FASTA format (default: None)
  --output-metadata OUTPUT_METADATA
                        metadata for strains that passed filters (default:
                        None)
  --output-strains OUTPUT_STRAINS
                        list of strains that passed filters (no header)
                        (default: None)
  --output-log OUTPUT_LOG
                        tab-delimited file with one row for each filtered
                        strain and the reason it was filtered. Keyword
                        arguments used for a given filter are reported in JSON
                        format in a `kwargs` column. (default: None)
  --empty-results-reporting {error,warn,skip}
                        How should empty filter results be reported. (default:
                        error)
```

Reply:
Looks good!  I like `--empty-results-reporting` over `--empty-output-reporting`.

Reply:
_non-blocking_

I think results vs. output is splitting hairs that and no one will be confused by it not applying to `--output-log`. We also don't use "results" anywhere in the `augur filter` help text and nearly nowhere else in Augur's user-facing interface, whereas we do use "output" everywhere for this concept.

Reply:
I like the resolution, the rewording of 

```
possible representations of filtered data (at least one required)
```
to 

```
options related to outputs, at least one of the possible representations
  of filtered data (--output, --output-metadata, --output-strains) is
  required`
```

is great!

Reply:
> We also don't use "results" anywhere in the `augur filter` help text and nearly nowhere else in Augur's user-facing interface, whereas we do use "output" everywhere for this concept.

This is a good point. I'm on board with `--empty-output-reporting` now.

Reply:
Thanks for the all the feedback! I updated the option to `--empty-output-reporting` and reworded the help message to be clear that the option applies to the filtered data:
```
  --empty-output-reporting {error,warn,silent}
                        How should empty outputs be reported when no strains
                        pass filtering and/or subsampling. (default: error)
```